app/lib/package/backend.dart

```diff
@@ -931,6 +967,7 @@ class PackageBackend {
     required User user,
     required PackageSummary archive,
     required String guid,
+    required bool hasCanonicalArchiveObject,
```

Comment:
This is just an optimization, right?

Because you don't want to overwrite if it already exists with the same bytes.
I think we can omit this optimization..? Won't it reduce the amount of code flows we have to test?

Reply:
Good question on the testability. I've added it to reduce the time we are spending in the transaction: it is a work that is a no-op in the end, so an extra flag won't really hurt to check for it. We can remove it now (or maybe later too)..